code/__HELPERS/turfs.dm

```diff
@@ -413,3 +413,17 @@ Turf and target are separate in case you want to teleport some distance from a t
 	if(pressure <= 20 || pressure >= 550)
 		return FALSE
 	return TRUE
+
+/// returns a turf that isn't holy from the list
+/proc/get_non_holy_tile_from_list(list/turf_list)
+	if(!length(turf_list))
+		CRASH("No turf list has been given")
+	var/list/copied_turf_list = turf_list.Copy()
+	var/turf/found_tile
+	do
+		found_tile = get_turf(pick_n_take(copied_turf_list))
+	while(found_tile.is_holy() && length(copied_turf_list))
+
+	if(found_tile.is_holy()) // we found no valid tile at all
+		return
+	return found_tile
```

Comment:
Am I reading this wrong? This seems to have the potential to be an infinite do while loop. 
Why not use a for loop with a definitive end?

Reply:
It's pick_and_take

Reply:
I added a comment so it will be clear

Reply:
:+1: